---
title: "Portfolio"
output:
  pdf_document:
    latex_engine: xelatex
---
# **Signaling pathways caused by exogenous carcinogens in vitro associated with oncogenesis**
### 2019250140 Jeon in a (BSMS222)


## 1. Introduction

### 1. 1. Background

Lung cancer remains the most common malignancy and the leading cause of cancer mortality worldwide. Lung cancer is known to be mainly caused by direct exposure to cigarettes. But LUAD (lung adenocarcinoma) in East Asia, especially in Taiwan, is characterized by a high rate of never-smokers, early onset, and predominant EGFR mutations (Chen et al., 2020). According to these analyses, APOBEC mutational signatures are frequently observed in younger females and enviromental carcinogen-like mutational signatures highly occur in older females. In addition, early onset is a distinct feature of LUAD in East Asia, especially among never-smokers. From this, we can think that LUAD is associated with genetic and environmental factors, especially in this paper, which analyzed that air pollution in Taiwan correlates with LUAD incidence in never-smokers.\n
The carcinogen signals presented in the paper include (1) Nitrosamine-like, (2) Nitro-PAH, (3) radiation, (4) Alkylating agents, and (5) PAHs. Among these five carcinogen, I will focus on Nitrosamine, Nitro-PAH, and PAHs.\n
Among the many components in tobacco smoke and outdoor and indoor air pollution are polycyclic aromatic hydrocarbons (PAHs), which are considered to be the most important carcinogens in these complex mixtures. Metabolism of PAHs leads to the formation of the active carcinogens. These reactive metabolites produce DNA adducts, resulting in DNA mutations, alteration of gene expression profiles, and tumorigenesis(Moorthy et al., 2015).\n
Nitrosamines are formed by a reaction between nitrates or nitrites and certain amines. Nitrosamines and/or their precursors can be found in diverse consumer products such as processed meats, alcoholic beverages, cosmetics, cigarette smoke and also be formed in the mouth or stomach if the food contains nitrosamine precursors. Nitrosamines are considered to be strong carcinogens that may produce cancer in diverse organs and tissues including lung, brain, liver, kidney, bladder, stomach, esophagus, and nasal sinus(H. Robles, 2014).\n
Nitrated polycyclic aromatic hydrocarbons (Nitro-PAHs) are derivatives of PAHs with at least one nitro-functional group (-NO2) on the aromatic ring. Nitro-PAHs are mainly generated by incomplete combustion and pyrolysis of fossil fuels and biomass. Nitro-PAHs are direct-acting mutagens and carcinogens. The mechanisms underlying some of these toxicological effects of nitro-PAHs include DNA damage, DNA adduct formation, aryl hydrocarbon receptor activation, changes in gene and protein expression, cell cycle alternations, increased levels of reactive oxygen species and pro-inflamation. Inhalation, oral ingestion and dermal contact are the main routes of nitro-PAH intake from the environment by humans and animals(Benjamin, 2017).


### 1. 2. Data Visualization Topic

The topic for data visualization is to plot the correlation between environmental carcinogen and enriched pathway. Particularly, focusing on the pathway that has a significant correlation with carcinogen, I will examine whether the carcinogens presented in the paper have a direct relationship with mutational metabolism.


## 2. Exploring Data

### 2. 1. Unboxing Dataset

Before drawing the plot, I loaded the packages needed to create a portfolio by using 'library()'.

```{r}
library(readxl)
library(tidyverse)
library(ggplot2)
library(dplyr)
library(rio)
library(ggridges)
library(cowplot)
```

I loaded the file by using 'rio' package. Plus, since the sheet to be used in this file is [4] "S5C_carcinogen 1Dpath_Fig5E", the data set was loaded using 'sheet'.

```{r}
d <- rio::import('https://ars.els-cdn.com/content/image/1-s2.0-S0092867420307431-mmc5.xlsx', sheet = 4)
```


### 2. 2. Manipulating Data Frame

After checking the data using 'head()', let's write 'colnames()' to check whether the column names are proper or not. 

```{r}
head(d)
colnames(d)
```

Since the column names are not organized, I set them as I wanted. In addition, row 1 and column 2, 3 are unnecessary, so I deleted them. I changed the columns except for the first column to numeric for facilitate processing, and rearranged the columns into a 'type'.

```{r}
d <- d[c(2:54), c(1, 4:18)]

d <- d %>%
  rename(pathway = ...1,
         PAHs_mRNA = PAHs,
         PAHs_prot = ...5,
         PAHs_phos = ...6,
         NitroPAHs_mRNA = NitroPAHs,
         NitroPAHs_prot = ...8,
         NitroPAHs_phos = ...9,
         Mixed_mRNA = Mixed,
         Mixed_prot = ...11,
         Mixed_phos = ...12,
         Nitrosamine_mRNA = Nitrosamine,
         Nitrosamine_prot = ...14,
         Nitrosamine_phos = ...15,
         mRNA = "Comparison of 6 carcinogen group (p-value)",
         prot = ...17,
         phos = ...18) %>%
  mutate(PAHs_mRNA = as.numeric(PAHs_mRNA),
         PAHs_prot = as.numeric(PAHs_prot),
         PAHs_phos = as.numeric(PAHs_phos),
         NitroPAHs_mRNA = as.numeric(NitroPAHs_mRNA),
         NitroPAHs_prot = as.numeric(NitroPAHs_prot),
         NitroPAHs_phos = as.numeric(NitroPAHs_phos),
         Mixed_mRNA = as.numeric(Mixed_mRNA),
         Mixed_prot = as.numeric(Mixed_prot),
         Mixed_phos = as.numeric(Mixed_phos),
         Nitrosamine_mRNA = as.numeric(Nitrosamine_mRNA),
         Nitrosamine_prot = as.numeric(Nitrosamine_prot),
         Nitrosamine_phos = as.numeric(Nitrosamine_phos),
         mRNA = as.numeric(mRNA),
         prot = as.numeric(prot),
         phos = as.numeric(phos)) %>% 
  gather(key = "type", value = "log", -c("pathway", "mRNA", "prot", "phos")) %>%
  gather("mRNA", "prot", "phos", key = "p_value", value = "P")
```

Let's check again to see whether it is changed properly.

```{r}
head(d)
```


## 3. Data Visualization

At first, I will draw Heatmap using the values of 'relative log2T/N', classified into mRNA, protein, and phosphate to see whether each carcinogens have significant effects on the enriched pathway. First, set the x-axis to 'pathway' and y-axis to 'log2T/N' values. Then write the code using 'geom_tile()' because I will draw a heatmap. At this time, the degree of log value will be compared, so write 'aes(fill=log)' in 'geom_tile().

```{r}
d %>% ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log))
```

Since 'pathway' letters on the x-axis overlap, let's adjust the angle and size so that the letters don't overlap.

```{r}
d %>% ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log)) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7))
```

Assign the color for good visibility. So that the carcinogen on the y-axis can be easily seen for each type. Also adjust the font size.

```{r}
d %>% ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log)) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7),
        axis.text.y = element_text(hjust = 1, size = 9,
                                   colour = c("black", "black", "black",
                                              "orange", "orange", "orange",
                                              "red", "red", "red",
                                              "limegreen", "limegreen", "limegreen")))
```

Change the color of heatmap to see the values easily, so the low-value was set to blue and the high-value to red. In this case, when the log value is 0, the value of Tumor and NAT is the same, and when the log value is negative, the NAT value is higher than Tumor, and vice versa. I set range using 'limits'. Values below 0 are marked in blue and over 0 are marked in red. The missing value 'NA' was marked in white.

```{r}
d %>% ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log)) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7),
        axis.text.y = element_text(hjust = 1, size = 9,
                                   colour = c("black", "black", "black",
                                              "orange", "orange", "orange",
                                              "red", "red", "red",
                                              "limegreen", "limegreen", "limegreen"))) +
  scale_fill_gradient2(midpoint = 0, low = "blue", high = "red",
                       limits = c(-1, 1.6), na.value = "white")
```

Name the title of the plot, the x- and y-axes, and the legend. Also change the position of legend to left.

```{r}
d %>% ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log)) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7),
        axis.text.y = element_text(hjust = 1, size = 9,
                                   colour = c("black", "black", "black",
                                              "orange", "orange", "orange",
                                              "red", "red", "red",
                                              "limegreen", "limegreen", "limegreen"))) +
  scale_fill_gradient2(midpoint = 0, low = "blue", high = "red",
                       limits = c(-1, 1.6), na.value = "white") +
  labs(title = "Correlation between Environmental Carcinogen and Enriched Pathway",
       cex.main = 8,
       x = "Carcinogen Enriched Pathway",
       y = "Type",
       fill = "log2 T/N") +
  theme(legend.position="left")
```

Since the title doesn't appear completely, adjust the size of the title.

```{r}

d %>% ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log)) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7),
        axis.text.y = element_text(hjust = 1, size = 9,
                                   colour = c("black", "black", "black",
                                              "orange", "orange", "orange",
                                              "red", "red", "red",
                                              "limegreen", "limegreen", "limegreen"))) +
  scale_fill_gradient2(midpoint = 0, low = "blue", high = "red",
                       limits = c(-1, 1.6), na.value = "white") +
  labs(title = "Correlation between \nEnvironmental Carcinogen and Enriched Pathway",
       cex.main = 8,
       x = "Carcinogen Enriched Pathway",
       y = "Type",
       fill = "log2 T/N") +
  theme(legend.position="left",
        plot.title = element_text(size=10, face="bold"))

```

Next, I will mark oultines to compare p-values to see if the hypothesis that carcinogen affects to the enriched pathway is significant.

> As our statistical hypothesis will, by definition, state some property of the distribution, the null hypothesis is the default hypothesis under which that property does not exist. This hypothesis might specify the probability distribution of X precisely, or it might only specify that it belongs to some class of distributions.
The p-value is used in the context of null hypothesis testing in order to quantify the statistical significance of a result, the result being the observed value of the chosen statistic T. The lower the p-value is, the lower the probability of getting that result if the null hypothesis were true. A result is said to be statistically significant if it allows us to reject the null hypothesis. All other things being equal, smaller p-values are taken as stronger evidence against the null hypothesis.

For typical analysis, using the standard α = 0.05 cutoff, the null hypothesis is rejected when p <= 0.05 and not rejected when p > 0.05. So, I marked p = 0.05 as a reference point for significance.

```{r}
d %>% filter(P <= 0.05) %>%
  ggplot(aes(x = reorder(pathway, -log, min), y = type)) +
  geom_tile(aes(fill = log, color = P), size = 0.5) +
  theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 7),
        axis.text.y = element_text(hjust = 1, size = 9,
                                   colour = c("black", "black", "black",
                                              "orange", "orange", "orange",
                                              "red", "red", "red",
                                              "limegreen", "limegreen", "limegreen"))) +
  scale_fill_gradient2(midpoint = 0, low = "blue", high = "red",
                       limits = c(-1, 1.6), na.value = "white") +
  scale_color_gradient2(midpoint = 0.05, low = "black", high = "white",
                        limits = c(0, 0.05), na.value = "white") +
  labs(title = "Correlation between \nEnvironmental Carcinogen and Enriched Pathway",
       cex.main = 8,
       x = "Carcinogen Enriched Pathway",
       y = "Type",
       fill = "log2 T/N",
       color = "P-value") +
  theme(legend.position="right",
        plot.title = element_text(size=10, face="bold"))
ggsave('Figure 1.pdf', width = 7, height = 5, dpi = 300)
```


Next, I will visualize what proteins are enriched in the pathway that five carcinogen groups affecting. For this, I loaded two files by using 'rio' package. After loading, check the data whether it is proper or not.

```{r}
e <- rio::import('https://ars.els-cdn.com/content/image/1-s2.0-S0092867420307431-mmc5.xlsx', sheet = 5)
f <- rio::import('https://ars.els-cdn.com/content/image/1-s2.0-S0092867420307431-mmc1.xlsx', sheet = 5)
head(e)
head(f)
```

The column names are proper, so I will extract some data that could be of use. I will use enriched pathways in data 'e', and log values of patient in data 'f'.

```{r}
e <- e[, c(2, 17:22)]
f <- f[, c(1, 4:93)]
head(e)
head(f)
```

Two data sets will be merged to draw the plots. For this, there will have to be some standard. In this case, gene name can be the 'merge point'. But each column names are different, so I will rename the column name and merge them. And for handling data easily, other column names are also changed and columns are rearranged in data 'e' and 'f'.

```{r}
e <- e %>% rename(Aryl = 'Aryl hydrocarbon receptor signalling',
                  Chem = 'Chemical carcinogenesis',
                  Drug = 'Drug metabolism - cytochrome P450',
                  ErbB = 'ErbB signaling pathway',
                  MAPK = 'MAPK signaling pathway')
f <- f %>% rename(Gene = gene) %>%
  gather(key = "patient", value = "log", -c("Gene"))

nd <- merge(e, f, by = "Gene")
nd
```

Because I will use log values data to see genes' upregulation or downregulation, 'geom_density_ridges_gradient' function will be used.
At first, use 'filter' function to extract 'Aryl Hydrocarbon Receptor Signalling'.

```{r}
nd %>% filter(Aryl == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x)))
```

Adjust theme and add quantile lines to see distribution of data, and also add vline to see the value 'zero'.

```{r}
nd %>% filter(Aryl == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed")
```

To improve readability, change the color and add titles.

```{r}
nd %>% filter(Aryl == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "Aryl Hydrocarbon Receptor Signalling")
```

Finally, modify the positions and some details of plot.

```{r}
nd %>% filter(Aryl == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "Aryl Hydrocarbon Receptor Signalling") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.key.height = unit(1, 'cm'))
```

At the same way, draw other plots and save them respectively.

```{r}
Aryl <- nd %>% filter(Aryl == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "Aryl Hydrocarbon Receptor Signalling") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.key.height = unit(1, 'cm'))

Chem <- nd %>% filter(Chem == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "Chemical Carcinogenesis") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.key.height = unit(1, 'cm'))

Drug <- nd %>% filter(Drug == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "Drug Metabolism \n- cytochrome 450") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.key.height = unit(1, 'cm'))

NSCLC <- nd %>% filter(NSCLC == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "Non-Small Cell Lung Cancer") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.key.height = unit(1, 'cm'))

ErbB <- nd %>% filter(ErbB == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "ErbB Signaling Pathway") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        legend.key.height = unit(1, 'cm'))

MAPK <- nd %>% filter(MAPK == 'v') %>%
  select(-c(2:7)) %>%
  ggplot(aes(log, Gene)) +
  geom_density_ridges_gradient(aes(fill = stat(x))) +
  theme_ridges(grid = F) +
  stat_density_ridges(quantile_lines = T, quantiles = c(0.25, 0.5, 0.75), alpha = 0.2) +
  geom_vline(xintercept = 0, col = "black", linetype = "dashed") +
  scale_fill_viridis_c(option = "turbo", limits = c(-5, 5)) +
  xlim(-5, 5) +
  labs(x = "log2T/N",
       y = "Gene",
       title = "MAPK Signaling pathway") +
  theme(axis.title.x = element_text(hjust = 0.5),
        axis.title.y = element_text(hjust = 0.5),
        plot.title = element_text(hjust = 0.5),
        axis.text.y = element_text(size = 8),
        legend.key.height = unit(1, 'cm'))

Aryl
Chem
Drug
NSCLC
ErbB
MAPK
```

Lastly, let's merge pathway plots together. 'plot_grid' function will be used.

```{r}
title <- ggdraw() +
  draw_label("Proteins Affecting to Enriched Pathway",
             fontface = "bold",
             size = 25,
             vjust = 0.5)

p <- plot_grid(Aryl + theme(legend.position = "none"),
               Chem + theme(legend.position = "none"),
               Drug + theme(legend.position = "none"),
               NSCLC + theme(legend.position = "none"),
               ErbB + theme(legend.position = "none"),
               MAPK + theme(legend.position = "none"),
               labels = c("A", "B", "C", "D", "E", "F"),
               label_size = 10)

ps <- plot_grid(title, p, ncol = 1, rel_heights = c(0.1, 2))

legend <- get_legend(Aryl + theme(legend.box.margin = margin(0, 0, 0, 12)))

plot_grid(ps, legend, rel_widths = c(3, 0.5))
ggsave('test.pdf', width = 17, height = 14)
```


## 4. Discussion

Figure A shows that PAHs and nitrosamine have a high mutational signature ratio in phosphate and NitroPAHs have a high proteinic mutation ratio. In addition, as we can see the p-value in first Figure, it was found that phosphate and protein affect pathway as carcinogen. According to the first and second Figure, tumors harboring PAH or nitro-PAH signatures showed significant enrichment for pathways associated with metabolism and detoxification of chemical carcinogens, including the AHR and Cytochrome P450 pathways, known to contribute to carcinogenesis by PAH. The nitro-PAH and nitrosamines-like groups were dominated by DNA repair, ERBB/MAPK pathway, and TLR/RIG-1 T-cell signaling, which potentially link to the tumor initiation, cell proliferation, EMT malignant progression, and immune modulation in early carcinogenesis(Chen et al., 2020).\n
Through this plotting, it was possible to determine that various carcinogens (PAHs, nitroPAHs, nitrosamine, etc.) absorbed into the body through smoking or air pollution affect to metabolism. Especially, the fact that experimental group used in data is never-smoker makes us to think environmental pollution is main carcinogen in TW cohort.


## 5. Feedback
At first, I loaded the file from website by using 'rio' package instead of loading from folder in my laptop. So others now can also access the original data easily.
And in the first plot, I stated before that the log value below zero is unnecessary information, but that opinion came from my ignorance. The less log value belows zero, the more downregulation it represents. Thus, I added negative log values in the plot.
When the R file was knitted to pdf, some long codes diplayed beyond the code block, so others couldn't see whole code. I used 'enter' buttons properly to improve readability.
And most importantly, I merged existing two plots together. I thought two Figures displayed same conclusion, so it wasn't meaningful. Instead of merging them, I added outer border to show p-value so we can understand which value represents significance.


## 6. Reference
Chen et al. (2020), Cell, Proteogenomics of Non-smoking Lung Cancer in East Asia Delineates Molecular Signatures of Pathogenesis and Progression\n

Moorthy et al. (2015), Toxicological Sciences, Polycyclic Aromatic Hydrocarbons: From Metabolism to Lung Cancer\n

H.Robles (2014), Encyclopedia of Toxicology\n

Benjamin A. Musa Bandowe et al. (2017), Science of The Total Environment, Nitrated polycyclic aromatic hydrocarbons (nitro-PAHs) in the environment – A review